### ISSUE TITLE
Feature Request: Transformation in "on" keyboard when joining dataframes

### ISSUE THREAD

**Issue by eoteroe** [JuliaData/DataFrames.jl#3363]
df1=DataFrame(N=["JOHN", "PETER"] ,value=[1,2])
df2=DataFrame(n=["john","peter"],value2=[3,4])

want somehting like: 
leftjoin( df1 , df2, on:[ (:N=>lowercase) => :n])

**Comment by bkamins** [comment]
The way we will likely implement it is a join allowing arbitrary matching function.

How performance-critical is your request? (i.e. do you need speed or having functionality there is enough?)

**Comment by eoteroe** [comment]
Most of the cases I need it this, it is functionality the principal request. Thank you for your consideration.

**Comment by bkamins** [comment]
Yes, but we need to design API that is general enough (otherwise the number of things that users would need to learn would be too large).

**Comment by eoteroe** [comment]
Excellent Bogumil.  I didt'n know about this package. I am glad Julia is getting noticed. Packages are increasing. Again, thank you for the consideration of this feature. I'd say DataFrames is one hook for many Data analysts who are trying Julia, and every improve you and the contributors make are really welcome.
